baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/ForbidJavaxParameterType.java

```diff
@@ -0,0 +1,131 @@
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.predicates.TypePredicate;
+import com.google.errorprone.predicates.TypePredicates;
+import com.google.errorprone.util.ASTHelpers;
+import com.palantir.errorprone.ForbidJavax;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Symbol.ClassSymbol;
+import com.sun.tools.javac.code.Symbol.MethodSymbol;
+import com.sun.tools.javac.code.Symbol.TypeSymbol;
+import com.sun.tools.javac.code.Symbol.VarSymbol;
+import com.sun.tools.javac.code.Type;
+import java.lang.annotation.Annotation;
+import java.util.List;
+import java.util.function.Predicate;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = BugPattern.SeverityLevel.ERROR,
+        summary = "Supplying an object which uses legacy javax types, such as javax.ws.rs to a\n"
+                + "method which requires newer jakarta types is a runtime error. This check ensures\n"
+                + "that you only supply proper types to these methods which generally just take an\n"
+                + "untyped Object. There is no auto-fix for this check, you must fix it manually")
+public final class ForbidJavaxParameterType extends BugChecker implements BugChecker.MethodInvocationTreeMatcher {
+
+    private static final Predicate<Symbol> HAS_JAXRS_ANNOTATION =
+            SymbolPredicates.hasAnnotationWithPackage("javax.ws.rs");
+
+    private static final TypePredicate IMPLEMENTS_FEATURE = TypePredicates.isDescendantOf("javax.ws.rs.core.Feature");
+
+    @Override
+    public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+        List<? extends ExpressionTree> arguments = tree.getArguments();
+        MethodSymbol methodSymbol = ASTHelpers.getSymbol(tree);
+        List<VarSymbol> typeParameters = methodSymbol.getParameters();
+
+        if (typeParameters == null
+                || typeParameters.isEmpty()
+                || arguments == null
+                || arguments.isEmpty()
+                || arguments.size() != typeParameters.size()) {
+            return Description.NO_MATCH;
+        }
+
+        for (int i = 0; i < typeParameters.size(); ++i) {
+            VarSymbol parameter = typeParameters.get(i);
+            ExpressionTree argument = arguments.get(i);
+
+            Annotation maybeRequireJakarta = parameter.getAnnotation(ForbidJavax.class);
```

Comment:
Thoughts on keeping `com.palantir.conjure.java.runtime:conjure-java-annotations` as a `testRuntimeOnly` dependency, and using the fqcn string to reference this annotation? In general I try to avoid adding dependencies to baseline for this sort of check because it can weigh down the build process.

Reply:
I'm good with that. And this check would just...not run if that annotation wasn't present?